polkadot/node/network/availability-distribution/src/requester/mod.rs

```diff
@@ -162,6 +267,62 @@ impl Requester {
 			self.add_cores(ctx, runtime, leaf, leaf_session_index, cores).await?;
 		}
 
+		if self.speculative_availability {
+			let scheduled_cores = self.request_backable_candidates_core_info(ctx, new_head).await?;
+			if scheduled_cores.len() > 0 {
+				gum::trace!(
+					target: LOG_TARGET,
+					?scheduled_cores,
+					"Query scheduled cores"
+				);
+
+				let candidate_hashes = scheduled_cores
+					.iter()
+					.map(|(_, core_info)| core_info.candidate_hash)
+					.collect::<HashSet<_>>();
+
+				let session_info = self
+					.session_cache
+					.get_session_info(
+						ctx,
+						runtime,
+						// We use leaf here, the relay_parent must be in the same session as
+						// the leaf. This is guaranteed by runtime which ensures that cores are
+						// cleared at session boundaries. At the same time, only leaves are
+						// guaranteed to be fetchable by the state trie.
+						leaf,
+						leaf_session_index,
+					)
+					.await
+					.map_err(|err| {
+						gum::warn!(
+							target: LOG_TARGET,
+							error = ?err,
+							"Failed to spawn a fetch task"
```

Comment:
```suggestion
                            session_index = ?leaf_session_index,
							"Failed to retrieve session info",
```

Reply:
I copied this from [somewhere else](https://github.com/paritytech/polkadot-sdk/blob/d8e7ba2ebb635b56f71eb5cc88c7a61d4f4a91d6/polkadot/node/network/availability-distribution/src/requester/mod.rs#L397) in the file with the same log.